snappy/buy.go

```diff
@@ -0,0 +1,39 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snappy
+
+import (
+	"fmt"
+	"github.com/ubuntu-core/snappy/progress"
+	"github.com/ubuntu-core/snappy/store"
+)
+
+// Buy the given snap name.
+func Buy(fullName string, meter progress.Meter) error {
+	mStore := NewConfiguredUbuntuStoreSnapRepository()
+	state, redirect, err := mStore.Buy(fullName, store.BuyOptions{}, meter, nil)
```

Comment:
As we discussed, it should talk to snapd rather than going directly to the store. The sequence would look along the lines of:
* snap <= client.Snap(name)
* snap.Prices, otherwise error with "it's free"
* snap.MustBuy, otherwise error with "done before"
* ask user for confirmation of price
* client.Buy(&BuyOptions{name, agreedPrice, agreedCurrency, etc})

You said in our meeting today that there were certain aspects of buying that might require other kinds of interaction, etc. I'd be curious to hear about those before we finalize the shape of that Buy client command and respective daemon support.